common/common-api/src/main/java/org/ow2/proactive/scripting/Script.java

```diff
@@ -392,28 +384,30 @@ protected final void prepareBindings(Bindings bindings) {
 
     /** Create string script from url */
     protected void storeScript(URL url) throws IOException {
-        BufferedReader buf = new BufferedReader(new InputStreamReader(url.openStream()));
-        StringBuilder builder = new StringBuilder();
-        String tmp;
+        try (BufferedReader buf = new BufferedReader(new InputStreamReader(url.openStream()))) {
+            StringBuilder builder = new StringBuilder();
+            String tmp;
 
-        while ((tmp = buf.readLine()) != null) {
-            builder.append(tmp).append("\n");
-        }
+            while ((tmp = buf.readLine()) != null) {
+                builder.append(tmp).append("\n");
+            }
 
-        script = builder.toString();
+            script = builder.toString();
+        }
     }
 
     /** Create string script from file */
     public static String readFile(File file) throws IOException {
-        BufferedReader buf = new BufferedReader(new InputStreamReader(new FileInputStream(file)));
-        StringBuilder builder = new StringBuilder();
-        String tmp;
+        try (BufferedReader buf = new BufferedReader(new InputStreamReader(new FileInputStream(file)))) {
```

Comment:
to close automatically the stream when we exit the try and avoid keeping file handles.